File: zoomx/src/main/java/com/zoomx/zoomx/util/SharedPreferenceManager.java
package com.zoomx.zoomx.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by Ibrahim AbdelGawad on 12/19/2017.
 */

public class SharedPreferenceManager {
    private Context context;
    private SharedPreferences.Editor editor;
    private SharedPreferences appSharedPrefs;

    public SharedPreferenceManager(Context context) {
        this.context = context;
        this.appSharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
        this.editor = this.appSharedPrefs.edit();
    }

    public boolean getBoolean(String key) {
        return this.getBoolean(key, false);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return this.appSharedPrefs.getBoolean(key, defaultValue);
    }

    public long getLong(String key, long defaultValue) {
        return this.appSharedPrefs.getLong(key, defaultValue);
    }

    public long getLong(String key) {
        return this.appSharedPrefs.getLong(key, 0L);
    }

    public void saveBoolean(String key, boolean value) {
        this.editor.putBoolean(key, value).commit();
    }

    public String getString(String key) {
        return this.getString(key, (String) null);
    }

    public String getString(String key, String defaultValue) {
        return this.appSharedPrefs.getString(key, defaultValue);
    }

    public void saveString(String key, String value) {
        this.editor.putString(key, value).commit();
    }

    public int getInt(String key, int defaultVal) {
        return this.appSharedPrefs.getInt(key, defaultVal);
    }

    public void saveInt(String key, int value) {
        this.editor.putInt(key, value).commit();
    }

    public boolean contains(String key) {
        return this.appSharedPrefs.contains(key);
    }

    public void removeEntry(String key) {
        this.editor.remove(key).commit();
    }

    public void clearSharedPreferences() {
        this.appSharedPrefs.edit().clear().commit();
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/util/SharedPreferenceManager.kt
package com.zoomx.zoomx.util

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

/**
 * Created by Ibrahim AbdelGawad on 12/19/2017.
 */

class SharedPreferenceManager(private val context: Context) {
    private val editor: SharedPreferences.Editor
    private val appSharedPrefs: SharedPreferences

    init {
        this.appSharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
        this.editor = this.appSharedPrefs.edit()
    }

    fun getBoolean(key: String): Boolean {
        return this.getBoolean(key, false)
    }

    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return this.appSharedPrefs.getBoolean(key, defaultValue)
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return this.appSharedPrefs.getLong(key, defaultValue)
    }

    fun getLong(key: String): Long {
        return this.appSharedPrefs.getLong(key, 0L)
    }

    fun saveBoolean(key: String, value: Boolean) {
        this.editor.putBoolean(key, value).commit()
    }

    fun getString(key: String): String? {
        return this.getString(key, null as String?)
    }

    fun getString(key: String, defaultValue: String?): String? {
        return this.appSharedPrefs.getString(key, defaultValue)
    }

    fun saveString(key: String, value: String) {
        this.editor.putString(key, value).commit()
    }

    fun getInt(key: String, defaultVal: Int): Int {
        return this.appSharedPrefs.getInt(key, defaultVal)
    }

    fun saveInt(key: String, value: Int) {
        this.editor.putInt(key, value).commit()
    }

    operator fun contains(key: String): Boolean {
        return this.appSharedPrefs.contains(key)
    }

    fun removeEntry(key: String) {
        this.editor.remove(key).commit()
    }

    fun clearSharedPreferences() {
        this.appSharedPrefs.edit().clear().commit()
    }
}
